### ISSUE TITLE
Efficientnet inference time is very slow!!!

### ISSUE THREAD

**Issue by 2696120622** [tensorflow/tpu#837]
On windows/cpu, I test the inference time with efficientnet-b0 following eval_ckpt_example.ipynb. It costs 12 second for just one panda.jpg.  On the other hand, I test the foreground propagation time for effecientnet-lite0 and efficientnet-b0 as follows.
effecientnet-lite0: 0.055s  
efficientnet-b0: 0.081s  
The time is longer than claimed in this respository.
Why is efficientnet inference time slow?

**Comment by mingxingtan** [comment]
It is because your inference time include the graph building time. If you want to test the run time, you should do something like:

# build graph
input = tf.placeholder(....)
output = ...

# run graph
with tf.Session() as ss:
  # skip warmup
  ss.run(output)

  # time the latency here only.
  ss.run(output)

**Comment by 2696120622** [comment]
> It is because your inference time include the graph building time. If you want to test the run time, you should do something like:
> 
> ```
> # build graph
> input = tf.placeholder(....)
> output = ...
> 
> # run graph
> with tf.Session() as ss:
>   # skip warmup
>   ss.run(output)
> 
>   # time the latency here only.
>   ss.run(output)
> ```

Thanks a lot.
I have retest the inference time on windows/cpu.following your tips.
`import tensorflow as tf
import model_builder_factory
import numpy as np

print(tf.__version__)
print('efficientnet-b0')
tf.reset_default_graph()
X = tf.placeholder(tf.float32,(1,224,224,3))
is_training = False
model_builder = model_builder_factory.get_model_builder('efficientnet-b0')
logits, _ = model_builder.build_model(X, 'efficientnet-b0', is_training)
prob = probs = tf.nn.softmax(logits)
with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    for i in range(20):
        start = datetime.now()
        _ = sess.run(prob,feed_dict={X:np.random.normal(size=(1,224,224,3)).astype(np.float32)})
        print(datetime.now()-start)

print('--------------------')
print('efficientnet-lite0')
tf.reset_default_graph()
X = tf.placeholder(tf.float32,(1,224,224,3))
is_training = False
model_builder = model_builder_factory.get_model_builder('efficientnet-lite0')
logits, _ = model_builder.build_model(X, 'efficientnet-lite0', is_training)
prob = probs = tf.nn.softmax(logits)
with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    for i in range(20):
        start = datetime.now()
        _ = sess.run(prob,feed_dict={X:np.random.normal(size=(1,224,224,3)).astype(np.float32)})
        print(datetime.now()-start)
`

The inference time has decreased a lot. But, it is still longer than claimed in this respository. as follows:
1.15.0
efficientnet-b0
0:00:00.318850
0:00:00.057151
0:00:00.059156
0:00:00.056150
0:00:00.065174
0:00:00.058155
0:00:00.056148
0:00:00.055147
0:00:00.055147
0:00:00.053142
0:00:00.053141
0:00:00.049141
0:00:00.050133
0:00:00.053141
0:00:00.052138
0:00:00.056150
0:00:00.055146
0:00:00.061157
0:00:00.052139
0:00:00.050133
--------------------
efficientnet-lite0
0:00:00.181516
0:00:00.039102
0:00:00.039073
0:00:00.040108
0:00:00.041109
0:00:00.040106
0:00:00.047133
0:00:00.038094
0:00:00.038101
0:00:00.037099
0:00:00.037108
0:00:00.042112
0:00:00.038101
0:00:00.037098
0:00:00.038099
0:00:00.040107
0:00:00.042112
0:00:00.037099
0:00:00.038101
0:00:00.037099

**Comment by 2696120622** [comment]
After coverting the trained ckpt model to tflite model, I test the inference time for fp32 and int8. the int8 tflite model take more time than fp32 tflite model. And tf32 tflite model take more time than the ckpt model.

`from datetime import datetime
import numpy as np
print('lite0-fp32 inference time')
# import tflite_runtime.interpreter as tflite
interpreter = tf.lite.Interpreter(model_path="../efficientnet_ckpt_lite/efficientnet-lite0/efficientnet-lite0-fp32.tflite")
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()
for i in range(20):
    start = datetime.now()
    interpreter.set_tensor(input_details[0]['index'], np.ones((1,224,224,3),np.float32))
    interpreter.invoke()
    output_data = interpreter.get_tensor(output_details[0]['index'])
    print(datetime.now()-start)

print('====================')
print('lite0-int8 inference time')
# import tflite_runtime.interpreter as tflite
interpreter = tf.lite.Interpreter(model_path="../efficientnet_ckpt_lite/efficientnet-lite0/efficientnet-lite0-int8.tflite")
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()
for i in range(20):
    start = datetime.now()
    interpreter.set_tensor(input_details[0]['index'], np.ones((1,224,224,3),np.uint8))
    interpreter.invoke()
    output_data = interpreter.get_tensor(output_details[0]['index'])
    print(datetime.now()-start)`

lite0-fp32 inference time
0:00:00.106284
0:00:00.091215
0:00:00.092245
0:00:00.090241
0:00:00.089237
0:00:00.092248
0:00:00.089234
0:00:00.092246
0:00:00.089238
0:00:00.091230
0:00:00.089235
0:00:00.092275
0:00:00.088206
0:00:00.099264
0:00:00.098266
0:00:00.103275
0:00:00.099263
0:00:00.100259
0:00:00.096256
0:00:00.090239

====================


lite0-int8 inference time
0:00:02.812481
0:00:02.812517
0:00:02.918728
0:00:02.820502
0:00:02.775383
0:00:02.816508
0:00:02.829482
0:00:02.825545
0:00:02.828529
0:00:02.852583
0:00:02.779393
0:00:02.816464
0:00:02.800481
0:00:02.788414
0:00:02.806437
0:00:02.824513
0:00:02.817495
0:00:02.848605
0:00:02.812487
0:00:02.798420
